src/jit/lclvars.cpp

```diff
@@ -3687,63 +3685,66 @@ void Compiler::lvaMarkLclRefs(GenTree* tree, BasicBlock* block, GenTreeStmt* stm
         }
     }
 
-    /* Is this an assigment? */
-
-    if (tree->OperIsAssignment())
+    if (!isRecompute)
```

Comment:
Why is the earlier `if (tree->gtOper == GT_CALL && tree->gtFlags & GTF_CALL_UNMANAGED)` case not also under `if (!isRecompute)`?

I wonder if the code would be simpler if instead of introducing this `isRecompute" flag everywhere, if there was a completely separate code path for recomputation. This "initial" computation could potentially be given a different name, e.g., "IRPassOne", because it does a set of different things (not that I'm actually suggesting that name), and there could be a `lvaRecomputeRefCnts` that just did that.

Reply:
Well, the "moral equivalent" I was suggesting was a separate "main" function entrypoint.

Reply:
Let me get this change and the next bit through and then I'll reexamine and see if we can re-express things more cleanly.

Reply:
Added an note as such to #18969.